src/api/adamik/broadcast.ts

```diff
@@ -24,6 +31,7 @@ export const broadcast = async (
   const result = await response.json();
   if (response.status !== 200) {
     console.error("broadcast - backend error:", JSON.stringify(result));
+    throw new Error(JSON.stringify(result));
```

Comment:
Beware, I think throwing errors in server components (see `"use server"` at the top of this file) is silent in production. It's expected and there's a rationale for this, but I think I found out this is not the best way to handle errors from server components (rather just return `null`, or just the error string or status).

Not 100% sure, you could go ahead and see if it works in prod. But anyway as mentioned with the "TODO" in each file here, API error management on app side should be completely reworked...